# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLCanvasElement.h
# Commit: 6d6c78c31c28
# Full Hash: 6d6c78c31c28c6d6827abfb392ccab10a24381ad
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-12-10 09:10:34
# Regressor Bug: 1736177
# File Overlap Count: 2
# Description:
#   Bug 1736177 - Part 8. Remove unused visibility observer from HTMLCanvasElement. r=jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D130784
# ==============================================================================

diff -r d55f1ee88bb9 -r 6d6c78c31c28 dom/html/HTMLCanvasElement.h
--- a/dom/html/HTMLCanvasElement.h	Thu Dec 09 19:25:27 2021 +0000
+++ b/dom/html/HTMLCanvasElement.h	Thu Dec 09 19:25:27 2021 +0000
@@ -58,19 +58,14 @@
 
 // Listen visibilitychange and memory-pressure event and inform
 // context when event is fired.
-class HTMLCanvasElementObserver final : public nsIObserver,
-                                        public nsIDOMEventListener {
+class HTMLCanvasElementObserver final : public nsIObserver {
  public:
   NS_DECL_ISUPPORTS
   NS_DECL_NSIOBSERVER
-  NS_DECL_NSIDOMEVENTLISTENER
 
   explicit HTMLCanvasElementObserver(HTMLCanvasElement* aElement);
   void Destroy();
 
-  void RegisterVisibilityChangeEvent();
-  void UnregisterVisibilityChangeEvent();
-
   void RegisterObserverEvents();
   void UnregisterObserverEvents();
 
@@ -319,7 +314,6 @@
 
   layers::LayersBackend GetCompositorBackendType() const;
 
-  void OnVisibilityChange();
   void OnMemoryPressure();
   void OnDeviceReset();
 
